django/db/backends/utils.py

```diff
@@ -163,6 +232,55 @@ def debug_sql(
             )
 
 
+class AsyncCursorDebugWrapper(AsyncCursorWrapper):
+    # XXX callproc isn't instrumented at this time.
+
+    async def execute(self, sql, params=None):
+        with self.debug_sql(sql, params, use_last_executed_query=True):
+            return await super().execute(sql, params)
+
+    async def executemany(self, sql, param_list):
+        with self.debug_sql(sql, param_list, many=True):
+            return await super().executemany(sql, param_list)
+
+    @contextmanager
+    def debug_sql(
+        self, sql=None, params=None, use_last_executed_query=False, many=False
+    ):
+        start = time.monotonic()
+        try:
+            yield
+        finally:
+            stop = time.monotonic()
+            duration = stop - start
+            if use_last_executed_query:
+                sql = self.db.ops.last_executed_query(self.cursor, sql, params)
```

Comment:
Might this need to be async?

Reply:
`last_executed_query` is a non-blocking method that basically only does string interpolation / decoding of a string that's already stored on the DBAPI cursor.